File: client-norx/app/src/main/java/ua/in/meugen/myapplication/ui/activities/base/fragment/BaseFragmentModule.java
package ua.in.meugen.myapplication.ui.activities.base.fragment;

import android.support.v4.app.Fragment;

import dagger.Binds;
import dagger.Module;
import ua.in.meugen.myapplication.app.di.scopes.PerFragment;

@Module
public abstract class BaseFragmentModule {

    @Binds @PerFragment
    abstract Fragment bindFragment(final BaseFragment fragment);
}

File: client-coroutines/app/src/main/java/ua/meugen/android/client/coroutines/ui/activities/base/fragment/BaseFragmentModule.kt
package ua.meugen.android.client.coroutines.ui.activities.base.fragment

import android.support.v4.app.Fragment
import android.support.v4.app.LoaderManager
import dagger.Binds
import dagger.Module
import dagger.Provides
import ua.meugen.android.client.coroutines.app.di.PerFragment
import ua.meugen.android.client.coroutines.ui.lifecycle.LifecycleHandler
import ua.meugen.android.client.coroutines.ui.lifecycle.LoaderLifecycleHandler

@Module
abstract class BaseFragmentModule {

    @Module
    companion object {

        @JvmStatic @Provides @PerFragment
        fun provideLoaderManager(fragment: Fragment): LoaderManager {
            return fragment.loaderManager
        }
    }

    @Binds @PerFragment
    abstract fun bindLifecycleHandler(handler: LoaderLifecycleHandler): LifecycleHandler
}